Lib/pyclbr.py

```diff
@@ -160,17 +160,20 @@ def _readmodule(module, path, inpackage=None):
     else:
         search_path = path + sys.path
     spec = importlib.util._find_spec_from_path(fullmodule, search_path)
+    if spec is None:
+        raise ModuleNotFoundError(f"no module named {fullmodule!r}", name=fullmodule)
     _modules[fullmodule] = tree
     # Is module a package?
-    if spec.submodule_search_locations is not None:
+    if getattr(spec, 'submodule_search_locations', None) is not None:
         tree['__path__'] = spec.submodule_search_locations
     try:
         source = spec.loader.get_source(fullmodule)
-        if source is None:
-            return tree
     except (AttributeError, ImportError):
         # If module is not Python source, we cannot do anything.
         return tree
+    else:
+        if source is None:
+            return tree
```

Comment:
This one is mostly just personal preference, I think this indent/block is unneeded since the try-except handles the exception case by returning early and, if the else block were to be excluded, wouldn't affect the success case (I'm not suggesting exclude lines 175 & 176 but to deindent them and remove the `else`).

Reply:
And my personal preference is the way I did it. :) Either way it shouldn't be in the `try` block and that's the key thing.